Update README.md (#9)
scale-it/algo-builder-templates: README.md

> ![image](https://github.com/scale-it/algo-builder/blob/master/media/logo.svg) 

Reviewer: you should use a cdn link: https://gitcdn.link/

Author: okay

# Algo Builder Templates

Dapps templates for [Algo Builder](https://github.com/scale-it/algo-builder). Use these templates to quickly setup/get started with your Dapp implementation.

Distributed Applications or dApps are basically decentralized applications. The templates in this repository are extremely helpful and easy to use. They are designed and implemented in such a way that they can provide the aspiring developers a headstart in building dApps based on `Algorand Blockchain`. The templates heavily use [algosdk](https://github.com/algorand/js-algorand-sdk), [algo-builder](https://github.com/scale-it/algo-builder) and [Algosigner](https://github.com/PureStake/algosigner).

The detailed description about each template and how t prperly use them can be found in their respective folders.

## Usage
Each template is a single project (with 1 package.json file) comprising of the following:

>  - An `algob` project : Used to deploy ASA's, stateless and stateful contracts using algob `deployer` object. Use `algob deploy` to deploy your `/scripts`. Sample algob project can  be found [here](https://github.com/scale-it/algo-builder/tree/master/packages/algob/sample-project), check the project `README.md` for more details.

Reviewer: please use a link for `README.md` instead of code marks.

Author: okay

   Read more about `@algo-builder/algob` [here](https://github.com/scale-it/algo-builder/tree/master/packages/algob).
 - [create-react-app](https://github.com/facebook/create-react-app): React js application for your frontend of web dApp. You can use the `AlgoSigner` global object in your app to use it's API's. Example can be found in `/bare` template.

   Deployment information (using `algob deploy` above) is stored in  `checkpoints` (in `/artifacts`) from which user can import information directly (loading `.yaml` file in react-app).

The templates can be easily imported or unboxed using the `algob unbox-template` command. To know how to use `algob` cli, visit [here](https://github.com/scale-it/algo-builder#installation) or simply run the following commands :

```bash

yarn global add "@algo-builder/algob"

> export PATH="$(yarn global bin):$PATH"

Reviewer: add a note that this should be added to the .profile file.

Author: okay

```

> The tutorial to use the command is as follows:

Reviewer: ```suggestion
## Quick Start
```

`algob unbox-template <template-name> <destination-directory> --force (flag)`

>  - if `destination-directory` is not passed then `pwd` will be used

Reviewer: ```suggestion
 - if `destination-directory` is not passed then current directory will be used
```

Author: will do the change.

>  - if `template-name` is not passed, then by default template "/bare" is unboxed

Reviewer: we are renaming `/bare` to `/default`. So let's use the new name already.

Author: okay

 - if `--force` is passed, then files are overwritten
 - if `template-name` passed is not present as an existing template, the command provides an interactive way to choose from the existing templates.

>  - If `--force` is not used and there are conflicting files, then user is made to choose whether to overwrite those files or not.

Reviewer: merge this bullet point with the one 2 lines above.

![image](https://gitcdn.link/repo/scale-it/algo-builder/master/media/logo.svg) 
# Algo Builder Templates

Dapps templates for [Algo Builder](https://github.com/scale-it/algo-builder). Use these templates to quickly setup/get started with your Dapp implementation.

> Distributed Applications or dApps are basically decentralized applications. The templates in this repository are extremely helpful and easy to use. They are designed and implemented in such a way that they can provide the aspiring developers a headstart in building dApps based on `Algorand Blockchain`. The templates heavily use [algosdk](https://github.com/algorand/js-algorand-sdk), [algo-builder](https://github.com/scale-it/algo-builder) and [Algosigner](https://github.com/PureStake/algosigner).

Reviewer: ```suggestion
Distributed Applications (dApps) are basically decentralized applications. The templates in this repository are extremely helpful and easy to use. They are designed and implemented in such a way that they can provide the aspiring developers a headstart in building dApps based on `Algorand Blockchain`. The templates heavily use [algosdk](https://github.com/algorand/js-algorand-sdk), [algo-builder](https://github.com/scale-it/algo-builder) and [Algosigner](https://github.com/PureStake/algosigner).
```

Author: okay

> The detailed description about each template and how t prperly use them can be found in their respective folders.

Reviewer: ```suggestion
The detailed description about each template and how to properly use them can be found in their respective folders.
```

## Usage
Each template is a single project (with 1 package.json file) comprising of the following:
 - An `algob` project : Used to deploy ASA's, stateless and stateful contracts using algob `deployer` object. Use `algob deploy` to deploy your `/scripts`. Sample algob project can  be found [here](https://github.com/scale-it/algo-builder/tree/master/packages/algob/sample-project), check the project [README.md](https://github.com/scale-it/algo-builder/blob/master/packages/algob/sample-project/README.md) for more details.
   Read more about `@algo-builder/algob` [here](https://github.com/scale-it/algo-builder/tree/master/packages/algob).
 - [create-react-app](https://github.com/facebook/create-react-app): React js application for your frontend of web dApp. You can use the `AlgoSigner` global object in your app to use it's API's. Example can be found in `/bare` template.

>    Deployment information (using `algob deploy` above) is stored in  `checkpoints` (in `/artifacts`) from which user can import information directly (loading `.yaml` file in react-app).

Reviewer: ```suggestion
Deployment information (using `algob deploy` above) is stored in  `checkpoints` (in `/artifacts`) from which user can import information directly (loading `.yaml` file in react-app).
```

The templates can be easily imported or unboxed using the `algob unbox-template` command. To know how to use `algob` cli, visit [here](https://github.com/scale-it/algo-builder#installation) or simply run the following commands :

```bash
yarn global add "@algo-builder/algob"
```

Note: Please make sure that yarn bin directory exists in your `PATH` variable.

To do so, run the following command:

```bash
export PATH="$(yarn global bin):$PATH"
```

## Quick Start

`algob unbox-template <template-name> <destination-directory> --force (flag)`
 - if `destination-directory` is not passed then current directory will be used.
 - if `template-name` is not passed, then by default template "/default" is unboxed.

>  - if `--force` is passed, then files are overwritten. If it isn't passed, then user is made to choose whether to overwrite those files or not.

Reviewer 2: ```suggestion
 - if `--force` is passed, then files are overwritten. If it isn't passed, then for each conflicting file, user is made to choose whether to overwrite that file or not.
```

 - if `template-name` passed is not present as an existing template, the command provides an interactive way to choose from the existing templates.
 - The command also asks if the user wants to install the dependencies as a part of the current process.

> 

Reviewer 2: Please also add a note about `algob` installation here. User will need to add algob (either global/local installation before using the template)

Reviewer: I think it's better to add only a link instead of description who to install algob: https://github.com/scale-it/algo-builder/#requirements

Reviewer 2: We should add a note that user "will need to install/link algob" after `unbox-template`. We can just add a link about the installation details like you said.

Author: done.
